src/libasr/ASR.asdl

```diff
@@ -283,10 +283,11 @@ expr
     | ArrayReshape(expr array, expr shape, ttype type, expr? value)
 
     | BitCast(expr source, expr mold, expr? size, ttype type, expr? value)
-    | DerivedRef(expr v, symbol m, ttype type, expr? value)
+    | StructInstanceMember(expr v, symbol m, ttype type, expr? value)
+    | EnumMember(expr v, symbol m, ttype type, expr? value)
```

Comment:
I would still do `EnumClassMember`. I think the `class` and `instance` relationship is well established and people understand it. I had to lookup our discussion in this PR to figure out what `EnumMember` means.

Essentially, Struct, Enum, Union, etc. they all have a "class" and "instance" and one can access both (with different semantics and consequences). If you don't like the term "class" then let's figure out a different word, to make this clear.

This can be done in a subsequent PR.

Reply:
I opened up an issue about this at https://github.com/lcompilers/lpython/issues/1183.